### PR TITLE
Fixes nan value for powc of zero

### LINKED ISSUE #114: powc of zero returns nan
When I execute the following line
```
Complex64::from(0.).powc(Complex64::from(1.))
```
I get nan as the output.

Shouldn't this be zero instead of nan?

### PR DESCRIPTION
**Pull request by domna**
Fixes #114 

@cuviper I just added a check for zero here as I suggested in #114.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -373,6 +373,10 @@ impl<T: Float> Complex<T> {
         // = p^c e^(−d θ) (cos(c θ + d ln(ρ)) + i sin(c θ + d ln(ρ)))
         // = from_polar(p^c e^(−d θ), c θ + d ln(ρ))
         let (r, theta) = self.to_polar();
+
+        if r.is_zero() {
+            return Self::new(r, r);
+        }
```
Comment (konsumlamm) [inline src/lib.rs:379]: This is wrong if `exp` is also zero or infinity or NaN.

Comment (konsumlamm) [reply src/lib.rs:379]: Also if `exp.re` is negative, as that would be dividing by zero.

Comment (konsumlamm) [reply src/lib.rs:379]: `exp` is the second argument (the exponent), not an angle.

Comment (domna) [reply src/lib.rs:379]: ah sorry. My bad, I was just focussing on the line you commented on and totally forget about what we're doing here. You are completely right, this cases should be handled here, too. I'm thinking about something like checking `if r.is_zero() && exp.norm().is_finite() && !exp.norm().is_zero()`, which should work. But maybe there is a simpler and cheaper approach.

File: src/lib.rs [hunk 2]
```diff
@@ -1908,6 +1897,13 @@ pub(crate) mod test {
                 Complex::new(1.65826, -0.33502),
                 1e-5
             ));
+            let z = Complex::new(0.0, 0.0);
+            assert!(close(z.powc(b), z));
+            assert!(z.powc(Complex64::new(0., 0.)).is_nan());
+            assert!(z.powc(Complex64::new(0., INFINITY)).is_nan());
+            assert!(z.powc(Complex64::new(10., INFINITY)).is_nan());
+            assert!(z.powc(Complex64::new(INFINITY, INFINITY)).is_nan());
+            assert!(close(z.powc(Complex64::new(INFINITY, 0.)), z));
```
Comment (konsumlamm) [inline src/lib.rs:1906]:
```suggestion
            assert!(close(z.powc(Complex64::new(INFINITY, 0.)), z));
            assert!(z.powc(Complex64::new(-1., 0.)).is_infinite());
```

Comment (domna) [reply src/lib.rs:1906]: This one is a tricky case as it is actually `infinity + iNaN`, which returns false for `is_infinite()`. But I think the return value is fine though, so I would suggest checking
```
 assert!(z.powc(Complex64::new(-1., 0.)).re.is_infinite());
 assert!(z.powc(Complex64::new(-1., 0.)).im.is_nan());
```

Comment (domna) [reply src/lib.rs:1906]: @konsumlamm Would you be happy with these altered tests? Or do you think in this case the function should return something like `infinity + i infinity`?

Comment (konsumlamm) [reply src/lib.rs:1906]: Yes, I'm happy, feel free to resolve  this.

File: src/lib.rs [hunk 3]
```diff
@@ -1908,6 +1897,17 @@ pub(crate) mod test {
...
+            assert!(z.powc(Complex64::new(-1., 0.)).re.is_infinite());
+            assert!(z.powc(Complex64::new(-1., 0.)).im.is_nan());
+            assert!(z.powc(Complex64::new(-1., 0.)).re.is_infinite());
+            assert!(z.powc(Complex64::new(-1., 0.)).im.is_nan());
```
Comment (konsumlamm) [inline src/lib.rs:1910]: You added the same test twice.

Comment (domna) [inline src/lib.rs:1910]:
```suggestion
```

Comment (domna) [reply src/lib.rs]: whoops, sorry. Removed it

**Comment by domna** [comment]
Is there anything I can still do to get this merged?

File: src/lib.rs [hunk 4]
```diff
@@ -1908,6 +1897,15 @@ pub(crate) mod test {
                 Complex::new(1.65826, -0.33502),
                 1e-5
             ));
+            let z = Complex::new(0.0, 0.0);
+            assert!(close(z.powc(b), z));
+            assert!(z.powc(Complex64::new(0., 0.)).is_nan());
```
Comment (cuviper) [inline src/lib.rs:1902]: I think this result should be one -- in fact the standard `f64::powf(x, 0.0) == 1.0` for *any* base, even NaN! I'm going to fix both `powf` and `powc` here...

**Comment by cuviper** [comment]
Let me know what you think of that additional change for 0 exponents. I'll tweak the release date when we're ready.

File: src/lib.rs (hunk 4 above)
Comment (konsumlamm) [reply src/lib.rs]: That's not mathematically correct though, 0^0 is [indeterminate](https://en.wikipedia.org/wiki/Indeterminate_form).

Comment (cuviper) [reply src/lib.rs]: See the prior discussions in <https://github.com/rust-num/num-traits/pull/78> and <https://github.com/rust-num/num-traits/pull/79>. It's a pragmatic choice, especially to be consistent with `std`.

Comment (cuviper) [reply src/lib.rs]: That wikipedia article also says:
> Another example is the expression 0^0. Whether this expression is left undefined, or is defined to equal 1, depends on the field of application and may vary between authors. For more, see the article [Zero to the power of zero](https://en.wikipedia.org/wiki/Zero_to_the_power_of_zero).

**Comment by domna** [comment]
> Let me know what you think of that additional change for 0 exponents. I'll tweak the release date when we're ready.

I think it makes sense. So I'm fine with it.

File: RELEASES.md [hunk 5]
```diff
@@ -1,9 +1,19 @@
+# Release 0.4.4 (2023-08-13)
+
+- [Fixes nan value for `powc` of zero][116]
```
Comment (konsumlamm) [inline RELEASES.md:3]:
```suggestion
- [Fixes NaN value for `powc` of zero][116]
```

**Pull request merged**